pyScienceMode2/rehastim_interface.py

```diff
@@ -371,3 +367,527 @@ def get_motomed_torque(self) -> float:
+        it will convert this number and interpret it as channel 2 of 4 [0,3] for the yellow connector.
+
+        Parameters
+        ----------
+        no_channel : int
+            The channel number.
+
+        Returns
+        -------
+        channel and connector
+        """
+        channels = [
+            sciencemode.lib.Smpt_Channel_Red,
+            sciencemode.lib.Smpt_Channel_Blue,
+            sciencemode.lib.Smpt_Channel_Black,
+            sciencemode.lib.Smpt_Channel_White,
+        ]
+
+        connectors = [sciencemode.lib.Smpt_Connector_Yellow, sciencemode.lib.Smpt_Connector_Green]
+
+        # Determine the connector
+        connector_idx = (no_channel - 1) // 4
+        connector = connectors[connector_idx]
+
+        # Determine the channel
+        channel = channels[(no_channel - 1) % 4]
+
+        return channel, connector
+
+    #  Low level commands
+
+    def ll_init(self):
+        """
+        Initialize the lower level of the device. The low-level is used for defining a custom shaped pulse.
+        Each stimulation pulse needs to triggered from the computer.
+        """
+        ll_init = sciencemode.ffi.new("Smpt_ll_init*")
+        ll_init.high_voltage_level = (
+            sciencemode.lib.Smpt_High_Voltage_Default
+        )  # This switches on the high voltage source
+        ll_init.packet_number = self.get_next_packet_number()
+
+        if not sciencemode.lib.smpt_send_ll_init(self.device, ll_init):
+            raise RuntimeError("Low level initialization failed.")
+        self.log(
+            "Low level initialized",
+            "Command sent to rehastim: {}".format(self.TypeRehap24(sciencemode.lib.Smpt_Cmd_Ll_Init).name),
+        )
+
+        self.get_next_packet_number()
+        self._get_last_ack()
+        self.check_ll_init_ack()
+
+    def check_ll_init_ack(self):
+        """
+        Check the low level initialization status.
+        """
+        if not sciencemode.lib.smpt_get_ll_init_ack(self.device, self.ll_init_ack):
+            raise RuntimeError("Low level initialization failed.")
+        generic_error_check(self.ll_init_ack, self.ERROR_MAP)
+
+    def start_ll_channel_config(
+        self,
+        no_channel,
+        points=None,
+        stim_sequence: int = None,
+        pulse_interval: int | float = None,
+        safety: bool = True,
+    ):
+        """
+        Starts the low level mode stimulation.
+
+        Parameters
+        ----------
+        no_channel : int
+            The channel number [1,8].
+        points : list
+            Points to stimulate. [1,16]
+        stim_sequence : int
+            Number of stimulation sequence to be repeated.
+        pulse_interval : int | float
+            Interval between each stimulation sequence in ms.
+        safety : bool
+            Set to True if you want to check the pulse symmetry. False otherwise.
+        """
+        self._current_no_channel = no_channel
```

Comment:
Check if channel number is an integer raise error if not

Reply:
Done.